geode-redis/src/main/java/org/apache/geode/redis/internal/executor/RedisKeyInRegion.java

```diff
@@ -102,4 +102,16 @@ protected RedisData getRedisDataOrDefault(ByteArrayWrapper key, RedisData defaul
       return result;
     }
   }
+
+  public boolean rename(ByteArrayWrapper oldKey, ByteArrayWrapper newKey) {
+    RedisData value = getRedisData(oldKey);
+    if (value == null) {
+      return false;
+    }
+
+    region.put(newKey, value);
+    region.remove(oldKey);
```

Comment:
If newKey is equal to oldKey it looks like this code will do the wrong thing. The redis rename docs say: "<= 3.2.0: Before Redis 3.2.0, an error is returned if source and destination names are the same.". Perhaps the best way to handle this is to check in the renameExecutor for equal keys and return "OK".

Reply:
we actually added some logic into the RedisKeyCommandsFunctionExecutor for if the key does not exist...we can probably add this here

Reply:
Updating this... are we concerned with Native Redis pre 3.2.0?

Reply:
Updating this and adding a test.